### PR TITLE
Fix request error on connection reset

### PR DESCRIPTION
**Pull request by kriskowal**
Revealed by testing against Hyperbahn.

r @jcorbin @ShanniLi 
cc @Raynos

### CODE REVIEW

File: node/connection_base.js [hunk 1]
```diff
@@ -175,7 +180,7 @@ TChannelConnectionBase.prototype.resetAll = function resetAll(err) {
         outPending: self.pending.out
     };
 
-    if (err.type.indexOf('tchannel.socket') < 0) {
+    if (err.type.indexOf('tchannel.socket', 0) < 0) {
```
Comment (Raynos) [inline node/connection_base.js]: why add the `0`; it should default to `0`.

File: node/request.js [hunk 2]
```diff
@@ -224,19 +224,36 @@ TChannelRequest.prototype.resend = function resend() {
...
+            onIdentified();
+        }
+    }
+
+    function onConnectionError(err) {
+        conn.closeEvent.removeListener(onConnectionError);
+        if (timer) {
+            self.timers.clearTimeout(timer);
+        }
+        self.emitError(err);
```
Comment (Raynos) [inline node/request.js]: `err || new Error('connection reset')`

We dont know if `err` will always exist or not.

File: node/connection_base.js (hunk 1 above)
Comment (kriskowal) [inline node/connection_base.js]: Oh, half done. Should be `lastIndexOf` to check for `/^tchannel.socket/`

**Comment by kriskowal** [comment]
@raynos Addressed feedback

**Comment by Raynos** [comment]
lgtm

**Comment by ShanniLi** [comment]
lgtm

**Comment by jcorbin** [comment]
:+1:

**Pull request merged**